pkg/server/plugin/nodeattestor/aws/session.go

```diff
@@ -35,22 +38,38 @@ func (cfg *SessionConfig) Validate(defaultAccessKeyID, defaultSecretAccessKey st
 }
 
 // newAWSSession create an AWS config from the credentials and given region
-func newAWSConfig(accessKeyID, secretAccessKey, region, assumeRoleArn string) aws.Config {
-	var credsProvider aws.CredentialsProvider
-	switch {
-	case assumeRoleArn != "":
-		stsConf := aws.Config{
-			Region: region,
+func newAWSConfig(accessKeyID, secretAccessKey, region, assumeRoleArn string) (aws.Config, error) {
+	ctx := context.Background()
+
+	var opts []func(*config.LoadOptions) error
+	if region != "" {
+		opts = append(opts, config.WithRegion(region))
+	}
+
+	if secretAccessKey != "" && accessKeyID != "" {
+		opts = append(opts, config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")))
+	}
+
+	stsConf, err := config.LoadDefaultConfig(ctx, opts...)
+	if err != nil {
+		return aws.Config{}, err
+	}
+
+	if assumeRoleArn != "" {
+		var opts []func(*config.LoadOptions) error
+		if region != "" {
+			opts = append(opts, config.WithRegion(region))
 		}
 
 		stsClient := sts.NewFromConfig(stsConf)
-		credsProvider = stscreds.NewAssumeRoleProvider(stsClient, assumeRoleArn)
-	case secretAccessKey != "" && accessKeyID != "":
-		credsProvider = aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, ""))
-	}
+		opts = append(opts, config.WithCredentialsProvider(aws.NewCredentialsCache(
+			stscreds.NewAssumeRoleProvider(stsClient, assumeRoleArn))),
+		)
 
-	return aws.Config{
-		Credentials: credsProvider,
-		Region:      region,
+		stsConf, err = config.LoadDefaultConfig(ctx, opts...)
+		if err != nil {
+			return aws.Config{}, err
+		}
 	}
```

Comment:
For readability, can we move this into a new helper method like `newAWSAssumeRoleConfig(region string, stsConf aws.Config)`? It gets a little hard to follow what options are being used where as currently written.